packages/core/types/src/schema/attribute/definitions/component.ts

```diff
@@ -47,4 +47,4 @@ export type GetComponentValue<TAttribute extends Attribute.Attribute> =
     : never;
 
 export type ComponentTarget<TAttribute extends Attribute.Attribute> =
-  TAttribute extends Component<infer TComponentUID> ? TComponentUID : never;
+  TAttribute extends Component<infer TComponentUID, Constants.BooleanValue> ? TComponentUID : never;
```

Comment:
can you remind me what that second param does ?

Reply:
It's used to indicate whether the component attribute is repeatable or not.
(https://github.com/strapi/strapi/blob/develop/packages/core/types/src/schema/attribute/definitions/component.ts#L18)

In this case, `TAttribute` was not matching the condition for repeatable components because the default value for `TRepeatable` is `false`, thus comparing `Component<UID, true>` and `Component<UID, false>` (always false).

By adding the `Constants.BooleanValue`, we're forcing the comparison between `Component<UID, true>` and `Component<UID, boolean>` (the first does extend the second)